src/components/SubjectPhasePicker/SubjectPhasePicker.tsx

```diff
@@ -243,15 +246,64 @@ const SubjectPhasePicker: FC<SubjectPhasePickerData> = ({
     );
   };
 
+  const ButtonFocusUnderline = styled(Svg)<{ $color: OakColorName }>`
+    color: ${(props) => props.$color};
+  `;
+
+  const viewButtonRef = useRef<HTMLButtonElement>(null);
+  const depsRef = useRef(
+    selectedSubject &&
+      selectedPhase &&
+      path &&
+      path.startsWith("/teachers/curriculum/"),
+  );
+
+  useEffect(() => {
+    let hideOuterBorders = false;
+    let phaseBackgroundEnabled = true;
+    let subjectBackgroundEnabled = true;
+
+    if (selectedSubject && selectedPhase) {
+      viewButtonRef.current?.focus();
+    }
+
+    if (showSubjects) {
+      hideOuterBorders = true;
+      phaseBackgroundEnabled = false;
+    }
+
+    if (showPhases) {
+      hideOuterBorders = true;
+      subjectBackgroundEnabled = false;
+    }
+
+    setDisplayNewBorders(hideOuterBorders);
+    setPhaseBackground(phaseBackgroundEnabled ? "white" : "grey1");
```

Comment:
Recommend using a state like `phaseEnabled` instead of `phaseBackground` to decouple logic and style.